barman/output.py

```diff
@@ -676,11 +676,20 @@ def result_list_backup(self, backup_info, backup_size, wal_size, retention_statu
             self.info(backup_info.backup_id)
             return
 
-        out_list = ["%s %s " % (backup_info.server_name, backup_info.backup_id)]
+        out_list = ["%s %s" % (backup_info.server_name, backup_info.backup_id)]
+
         if backup_info.backup_name is not None:
-            out_list.append("'%s' - " % backup_info.backup_name)
+            out_list.append(" '%s'" % backup_info.backup_name)
+
+        # Backup type can be either full, incremental or rsync
+        if backup_info.config.backup_method == "rsync":
```

Comment:
I can see you changed this, but this is still volatile as it keeps depending on the configuration.

See this as an example -- `support` server was originally configured with `rsync`:

```console
$ grep -i method /etc/barman.d/support.conf
#backup_method = rsync
backup_method = postgres
```

The `barman list-backup` command will wrongly show them as "full":

```console
$ barman list-backup support
support 20240725T183751 'rsync_incremental' - F - Thu Jul 25 16:37:55 2024 - Size: 235.2 MiB - WAL Size: 0 B
support 20240725T183740 'rsync_full' - F - Thu Jul 25 16:37:47 2024 - Size: 235.2 MiB - WAL Size: 32.0 MiB
```